taichi/backends/cc/runtime/base.h

```diff
@@ -83,6 +83,52 @@ static inline Ti_f32 Ti_rand_f32(void) {
   return (Ti_f32) drand48();  // [0.0, 1.0)
 }
 
+// Copied from Metal:
+typedef Ti_u8 *Ti_AdStackPtr;
+
+static inline Ti_u32 *Ti_ad_stack_n(Ti_AdStackPtr stack) {
+  return (Ti_u32 *)stack;
+}
+
+static inline Ti_AdStackPtr Ti_ad_stack_data(Ti_AdStackPtr stack) {
+  return stack + sizeof(Ti_u32);
+}
+
+static inline void Ti_ad_stack_init(Ti_AdStackPtr stack) {
+  Ti_u32 *n = Ti_ad_stack_n(stack);
+  Ti_i32 *data = (Ti_i32 *)Ti_ad_stack_data(stack);
+  data[0] = 0;
+  data[1] = 0;
+  *n = 1;
```

Comment:
After merging #1791: 
```suggestion
  *n = 0;
```

Note that zero-initialization is already handled in `Ti_ad_stack_push`.